src/Symfony/Component/Routing/Matcher/RedirectableUrlMatcher.php

```diff
@@ -27,9 +28,12 @@ public function match($pathinfo)
         try {
             $parameters = parent::match($pathinfo);
         } catch (ResourceNotFoundException $e) {
-            if ('/' === substr($pathinfo, -1) || !in_array($this->context->getMethod(), array('HEAD', 'GET'))) {
+            if ('/' === substr($pathinfo, -1)) {
                 throw $e;
             }
+            if (!in_array($this->context->getMethod(), array('HEAD', 'GET'))) {
+                throw new MethodNotAllowedException(array('GET'));
```

Comment:
I don't get it. When there is no route for the current uri path (ResourceNotFoundException) and the request method was POST, then it throws a MethodNotAllowedException? But we don't know at this point that a GET to the same URI would be allowed, do we?